cfgov/v1/migrations/0234_clean_orphaned_revisions.py

```diff
@@ -0,0 +1,39 @@
+# Generated by Django 3.2.17 on 2023-03-14 13:09
+import logging
+
+from django.db import migrations
+
+
+logger = logging.getLogger(__name__)
+
+
+def forwards(apps, schema_editor):
+    Page = apps.get_model('wagtailcore', 'Page')
+    PageRevision = apps.get_model('wagtailcore', 'PageRevision')
+
+    for revision_pk, page_pk in PageRevision.objects.values_list('pk', 'page'):
+        try:
+            page = Page.objects.get(pk=page_pk)
+        except Page.DoesNotExist:
+            page = None
+        else:
+            continue
+
+        revision = PageRevision.objects.get(pk=revision_pk)
+        revision.page = Page(title="Temporary page")
+        logger.info(
+            f"Deleting orphaned page revision with pk {revision_pk} "
+            f"page pk {page_pk}, and path {revision.content['url_path']}."
+        )
+        revision.delete()
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('v1', '0233_crc_tables'),
+    ]
+
+    operations = [
+        migrations.RunPython(forwards),
```

Comment:
Might as well? We don't have a good convention about this but adding this does no harm and makes the migration reversible.

```suggestion
        migrations.RunPython(forwards, migrations.RunPython.noop),
```